src/Compilers/VisualBasic/Portable/Utilities/FeatureUtils.vb

```diff
@@ -0,0 +1,53 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Option Compare Binary
+Option Strict On
+
+Imports Microsoft.CodeAnalysis.VisualBasic
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
+
+    Friend Class FeatureUtils
+
+        Private Shared Function CheckLanguageVersion(languageVersion As LanguageVersion, feature As Feature) As Boolean
+            Dim required = feature.GetLanguageVersion()
+            Return required <= languageVersion
+        End Function
+
+        Protected Friend Shared Function ReportFeatureUnavailable(Of TNode As VisualBasicSyntaxNode)(feature As Feature, node As TNode, languageVersion As LanguageVersion) As TNode
```

Comment:
Keep the scope as narrow as possible, and not public so a public api review isn't needed.